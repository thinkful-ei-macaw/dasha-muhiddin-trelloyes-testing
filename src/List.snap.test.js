import React from "react";
import renderer from "react-test-renderer";

import List from "./List";

describe("<List />", () => {
  it("renders this UI as expected", () => {
    // Render the component, as JSON
    const tree = renderer
      .create(
        <List
          header="First list"
          cards={[
            { id: 1, title: "First card", content: "Lorem ipsum" },
            { id: 2, title: "Second card", content: "Lorem ipsum bla bla" }
          ]}
        />
      )
      .toJSON();
    // Check whether it matches the previous snapshot
    // Stored in __snapshots__/App.test.js.snap
    expect(tree).toMatchSnapshot();
  });
});
