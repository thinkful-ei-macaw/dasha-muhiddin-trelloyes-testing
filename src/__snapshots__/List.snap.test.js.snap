// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders this UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      First list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        Lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Second card
      </h3>
      <p>
        Lorem ipsum bla bla
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
